lib/elixir/lib/macro.ex

```diff
@@ -2642,8 +2642,23 @@ defmodule Macro do
     end
   end
 
+  defp dbg_ast_to_debuggable({:if, _meta, [condition_ast, _clauses]} = ast, env) do
+    case Macro.Env.lookup_import(env, {:if, 2}) do
+      [macro: Kernel] ->
+        quote do
+          condition_result = unquote(condition_ast)
+          result = if condition_result, unquote(clauses)
+
+          {:if, unquote(escape(ast)), unquote(escape(condition_ast)), condition_result, result}
```

Comment:
This fixes the bootstrapping issue:

```suggestion
  defp dbg_ast_to_debuggable({:if, meta, [condition_ast, clauses]} = ast, env) do
    case Macro.Env.lookup_import(env, {:if, 2}) do
      [macro: Kernel] ->
        condition_result_var = unique_var(:condition_result, __MODULE__)

        quote do
          unquote(condition_result_var) = unquote(condition_ast)
          result = unquote({:if, meta, [condition_result_var, clauses]})

          {:if, unquote(escape(ast)), unquote(escape(condition_ast)),
           unquote(condition_result_var), result}
```